063-topic-replication-factor-change.md

```diff
@@ -0,0 +1,271 @@
+
+The Topic Operator needs to authenticate and send HTTP requests to the Cruise Control's REST API.
+The following environment variables are set by the Cluster Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
+
+| Name                                   | Type    | Default | Description                                                   |
+|:---------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED         | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_NAMESPACE       | String  | ""      | Namespace where Cruise Control is running                     |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME        | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT            | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED     | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED    | Boolean | false   | Whether Basic authentication is enabled                       |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED    | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_SECRET_NAME     | String  | ""      | Secret name containing the truststore                         |
+| STRIMZI_CRUISE_CONTROL_API_SECRET_NAME | String  | ""      | Secret name containing the API credentials                    |
+
+With Cluster Operator deployment, a dedicated admin user is automatically created for the Topic Operator.
+With standalone deployment, the Cluster Operator expects two secrets that needs to be created by the user:
+
+- A secret with `cruise-control.p12` key containing the truststore in PKCS12 format, and `cruise-control.password` key containing the truststore password.
+- A secret with `cruise-control.apiToAdminUsername` key containing the API username with admin role, and `cruise-control.apiToAdminPassword` key containing the user password.
+
+### KafkaTopic CRD changes
+
+During the whole replicas change process the KafkaTopic remains in a Ready state.
+A replicas change is executed asynchronously, and takes more than one reconciliation to complete.
+
+A new optional KafkaTopicStatus subsection called `replicasChange` is used to update the user, and keep track of the replicas change state.
+At any given time, the replicas change can be pending, ongoing, or completed.
+
+- **Pending**: Not in Cruise Control's task queue (not yet sent or request error).
+  ```sh
+  status:
+    conditions:
+    - lastTransitionTime: "2024-01-18T16:13:50.490918232Z"
+      status: "True"
+      type: Ready
+    observedGeneration: 2
+    replicasChange:
+      completed: false
```

Comment:
Not a blocker. But does it make sense to have a `completed` field that seems to have only one option of `false`? Or when it would be `true`? Would it make more sense to track the state there? E.g. `state: Pending`, state: Ongoing` etc.? It might give you more flexibility if in the future you need to do something more in these states as well.

Reply:
I agree that it's better to record the resource states directly, rather than just the boolean, and also describe how these states map to the CC states (they may or may not be the same, depending how how things work).

Reply:
This is a good idea and something that I also tried at some point in my draft implementation and lost while working on status updates. I'm adding it back. 

The state will show pending and then ongoing. It won't show the completed state to avoid confusion. I will simply remove the entire `replicasChange` subsection when done.